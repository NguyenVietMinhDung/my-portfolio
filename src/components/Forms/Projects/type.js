import type { Props as Items } from './components/RelevantProjects/type';

export type Props = {
  title: string,
  items: Items,
};
