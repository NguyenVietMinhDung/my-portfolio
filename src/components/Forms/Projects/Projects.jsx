// @flow
import React, { memo } from 'react';
import CSSModules from 'react-css-modules';
import styles from './Projects.scss';
import type { Props } from './type';
import RelevantProjects from './components/RelevantProjects';

const Projects = (props: Props) => {
  const { title, items } = props;
  return (
    <div
      id="relevant-projects"
      styleName="background"
    >
      <div
        styleName="container"
        data-aos="fade-right"
        data-aos-duration="2000"
      >
        <div styleName="title">{title}</div>
        <div styleName="content">
          <RelevantProjects items={items} />
        </div>
      </div>
    </div>
  );
};

export default memo<Props>(CSSModules(Projects, styles));
