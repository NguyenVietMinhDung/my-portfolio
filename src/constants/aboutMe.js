export const TITLE = 'About Me';
export const GREETING = 'Hi There! I\'m Dung Nguyen';
export const SUMMARY = 'I am an ambitious Web\n'
  + 'Application Developer who\n'
  + 'always tries to improve himself\n'
  + 'everyday, a quick learner who\n'
  + 'can absorb new ideas and can\n'
  + 'communicate clearly and\n'
  + 'effectively. Looking for an\n'
  + 'ambitious company which will\n'
  + 'challenge my developer and\n'
  + 'allows me to continue\n'
  + 'to develop my knowledge and\n'
  + 'potential.';
export const basicInfo = [
  {
    iconClassName: 'lni-calendar',
    name: 'Date of birth',
    value: '16 January 1995',
  },
  {
    iconClassName: 'lni-envelope',
    name: 'Email',
    value: 'ngvietminhdung@gmail.com',
  },
  {
    iconClassName: 'lni-phone-handset',
    name: 'Phone',
    value: '(076) 401-2912',
  },
  {
    iconClassName: 'lni-map-marker',
    name: 'Location',
    value: 'Tan Binh District, HCM City',
  },
];
export const skills = [
  {
    itemId: 'skill-1',
    name: 'JavaScript, HTML, CSS',
    value: 80,
  },
  {
    itemId: 'skill-2',
    name: 'ReactJS',
    value: 60,
  },
  {
    itemId: 'skill-3',
    name: 'SCSS',
    value: 60,
  },
  {
    itemId: 'skill-5',
    name: 'Webpack',
    value: 50,
  },
  {
    itemId: 'skill-6',
    name: 'Node.js',
    value: 60,
  },
  {
    itemId: 'skill-7',
    name: 'Git',
    value: 60,
  },
];
