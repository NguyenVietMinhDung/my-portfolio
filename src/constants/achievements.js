export const TITLE = 'Achievements';
export const timeline = {
  iconClassName: 'lni-certificate',
  items: [
    {
      key: 1,
      organization: 'Amazon Web Services (AWS)',
      startDate: 'August 2020',
      endDate: 'August 2023',
      description: 'AWS Certified Cloud Practitioner',
    },
  ],
};
