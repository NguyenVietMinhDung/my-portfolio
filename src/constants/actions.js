export const ACTIVATE_NAVIGATION_ITEM = 'ACTIVATE_NAVIGATION_ITEM';
export const OPEN_NAVIGATION_MENU = 'OPEN_NAVIGATION_MENU';
