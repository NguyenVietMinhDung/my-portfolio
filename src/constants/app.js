import facebookIconUrl from '../assets/images/facebook.png';
import instagramIconUrl from '../assets/images/instagram.png';
import linkedInIconUrl from '../assets/images/linkedin.png';

export const LOADER_SIZE = 50;
export const LOADER_SIZE_UNIT = 'px';
export const LOADER_COLOR = '#754ef9';
export const socialNetworks = [
  {
    name: 'Facebook',
    url: 'https://www.facebook.com/minh.dung.391',
    iconUrl: facebookIconUrl,
  },
  {
    name: 'Instagram',
    url: 'https://www.instagram.com/dungvmnguyen',
    iconUrl: instagramIconUrl,
  },
  {
    name: 'LinkedIn',
    url: 'https://www.linkedin.com/in/dungvmn1995',
    iconUrl: linkedInIconUrl,
  },
];
export logoUrl from '../assets/images/logo.png';
