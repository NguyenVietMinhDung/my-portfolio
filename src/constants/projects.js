export const TITLE = 'Relevant Projects';
export const items = [
  {
    isCurrentProject: false,
    name: 'Textme',
    role: 'Developer',
    startDate: 'September 2019',
    endDate: 'May 2020',
    description: 'A chat tool that helps customers can contact with bankers and versa vice',
  },
  {
    isCurrentProject: false,
    name: 'Huddle',
    role: 'Developer',
    startDate: 'January 2019',
    endDate: 'June 2019',
    description: 'A ticket-selling website for US high school events',
  },
  {
    isCurrentProject: false,
    name: 'ERTPMTool',
    role: 'Developer',
    startDate: 'July 2018',
    endDate: 'January 2019',
    description: 'A micro-services project management application\n'
    + 'which helps managers to keep track of resource allocation',
  },
  {
    isCurrentProject: false,
    name: 'PDV',
    role: 'Developer',
    startDate: 'February 2018',
    endDate: 'June 2018',
    description: 'Provider Search is a user interface that allows\n'
    + 'users to search for practitioners in the MasterFile. Built on the Single',
  },
  {
    isCurrentProject: false,
    name: 'DMTool',
    role: 'Developer',
    startDate: 'September 2017',
    endDate: 'February 2018',
    description: 'An internal tool which supports managers to keep track of resource allocation',
  },
];
