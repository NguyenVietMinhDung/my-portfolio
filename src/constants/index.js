import * as actions from './actions';
import * as app from './app';
import * as navbar from './navbar';
import * as header from './header';
import * as footer from './footer';
import * as aboutMe from './aboutMe';
import * as experiences from './experiences';
import * as educations from './educations';
import * as projects from './projects';
import * as achievements from './achievements';

export default {
  actions,
  app,
  navbar,
  header,
  footer,
  aboutMe,
  experiences,
  educations,
  projects,
  achievements,
};
