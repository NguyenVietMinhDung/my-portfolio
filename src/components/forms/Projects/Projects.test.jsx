import React from 'react';
import { shallow } from 'enzyme';
import { TITLE, items } from '../../../constants/projects';
import Projects from './Projects';

const defaultProps = {
  title: TITLE,
  items,
};

describe('<Projects />', () => {
  test('renders correctly', () => {
    const component = shallow(<Projects {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });
});
