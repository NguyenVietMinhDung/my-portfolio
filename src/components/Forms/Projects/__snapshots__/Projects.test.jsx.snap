// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Projects /> renders correctly 1`] = `
<div
  className="background"
  id="relevant-projects"
>
  <div
    className="container"
    data-aos="fade-right"
    data-aos-duration="2000"
  >
    <div
      className="title"
    >
      Relevant Projects
    </div>
    <div
      className="content"
    >
      <Memo(RelevantProjects)
        items={
          Array [
            Object {
              "description": "A chat tool that helps customers can contact with bankers and versa vice",
              "endDate": "May 2020",
              "isCurrentProject": false,
              "name": "Textme",
              "role": "Developer",
              "startDate": "September 2019",
            },
            Object {
              "description": "A ticket-selling website for US high school events",
              "endDate": "June 2019",
              "isCurrentProject": false,
              "name": "Huddle",
              "role": "Developer",
              "startDate": "January 2019",
            },
            Object {
              "description": "A micro-services project management application
which helps managers to keep track of resource allocation",
              "endDate": "January 2019",
              "isCurrentProject": false,
              "name": "ERTPMTool",
              "role": "Developer",
              "startDate": "July 2018",
            },
            Object {
              "description": "Provider Search is a user interface that allows
users to search for practitioners in the MasterFile. Built on the Single",
              "endDate": "June 2018",
              "isCurrentProject": false,
              "name": "PDV",
              "role": "Developer",
              "startDate": "February 2018",
            },
            Object {
              "description": "An internal tool which supports managers to keep track of resource allocation",
              "endDate": "February 2018",
              "isCurrentProject": false,
              "name": "DMTool",
              "role": "Developer",
              "startDate": "September 2017",
            },
          ]
        }
      />
    </div>
  </div>
</div>
`;
